data/data-pipeline/data_pipeline/etl/sources/census_acs/etl_imputations.py

```diff
@@ -84,50 +73,56 @@ def calculate_income_measures(
-    tract_list, geo_df = _prepare_dataframe_for_imputation(
-        impute_var_named_tup_list=impute_var_named_tup_list,
+    tract_list = _get_impute_tract_list(
+        columns_to_impute=raw_fields,
         geo_df=geo_df,
         geoid_field=geoid_field,
     )
 
-    # Iterate through the dataframe to impute in place
-    ## TODO: We should probably convert this to a spatial join now that we are doing >1 imputation and it's taking a lot
-    ## of time, but thinking through how to do this while maintaining the masking will take some time. I think the best
-    ## way would be to (1) spatial join to all neighbors, and then (2) iterate to take the "smallest" set of neighbors...
-    ## but haven't implemented it yet.
-    for index, row in geo_df.iterrows():
-        if row[geoid_field] in tract_list:
-            neighbor_mask = _get_neighbor_mask(geo_df, row)
-            county_mask = _get_fips_mask(
-                geo_df=geo_df, row=row, fips_digits=5, geoid_field=geoid_field
-            )
-            ## TODO: Did CEQ decide to cut this?
-            state_mask = _get_fips_mask(
-                geo_df=geo_df, row=row, fips_digits=2, geoid_field=geoid_field
-            )
-
-            # Impute fields for every row missing at least one value using the best possible set of neighbors
-            # Note that later, we will pull raw.fillna(imputed), so the mechanics of this step aren't critical
-            for impute_var_pair in impute_var_named_tup_list:
-                mask_to_use = _choose_best_mask(
-                    geo_df=geo_df,
-                    masks_in_priority_order=[
-                        neighbor_mask,
-                        county_mask,
-                        state_mask,
-                    ],
-                    column_to_impute=impute_var_pair.raw_field_name,
-                )
-                geo_df.loc[index, impute_var_pair.imputed_field_name] = geo_df[
-                    mask_to_use
-                ][impute_var_pair.raw_field_name].mean()
-
-    logger.info("Casting geodataframe as a typical dataframe")
+    missing_tracts_df = geo_df[geo_df[GEOID_TRACT_FIELD].isin(tract_list)]
+
+    # Perform as spatial merge to save time
+    spatially_joined_df = missing_tracts_df[
+        [GEOID_TRACT_FIELD, "geometry"]
+    ].sjoin(
+        geo_df[raw_fields + ["geometry"]],
+        predicate="touches",
+    )
+
+    # First take the neighbor averages
+    neighbor_averages_df = (
+        spatially_joined_df.groupby(GEOID_TRACT_FIELD)[raw_fields]
+        .mean()
+        .reset_index()
+        .rename(columns=rename_dict)
+    )
+
+    logger.info("Merging and casting geodataframe as a typical dataframe")
     # get rid of the geometry column and cast as a typical df
     df = pd.DataFrame(
-        geo_df[[col for col in geo_df.columns if col != "geometry"]]
+        geo_df[[col for col in geo_df.columns if col != "geometry"]].merge(
```

Comment:
any reason to do the projection here and not just `geo_df.drop(columns='geometry')).merge`